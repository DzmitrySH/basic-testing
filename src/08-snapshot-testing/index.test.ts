import { generateLinkedList } from './index';

describe('generateLinkedList', () => {
  const values1 = ['L', 'i', 's', 't'];
  const expectedList = {
    value: 'L',
    next: {
      value: 'i',
      next: {
        value: 's',
        next: {
          value: 't',
          next: {
            value: null,
            next: null,
          },
        },
      },
    },
  };
  // Check match by expect(...).toStrictEqual(...)
  test('should generate linked list from values 1', () => {
    const result = generateLinkedList(values1);
    expect(result).toStrictEqual(expectedList);
  });

  // Check match by comparison with snapshot
  test('should generate linked list from values 2', () => {
    const values2 = ['L', 'i', 'n', 'k', 'e', 'd'];
    const result = generateLinkedList(values2);
    expect(result).toMatchSnapshot();
  });
});
