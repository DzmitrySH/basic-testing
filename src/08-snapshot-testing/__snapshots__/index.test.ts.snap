// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLinkedList should generate linked list from values 2 1`] = `
{
  "next": {
    "next": {
      "next": {
        "next": {
          "next": {
            "next": {
              "next": null,
              "value": null,
            },
            "value": "d",
          },
          "value": "e",
        },
        "value": "k",
      },
      "value": "n",
    },
    "value": "i",
  },
  "value": "L",
}
`;
